ocaml/xapi/xapi_globs.ml

```diff
@@ -34,7 +34,13 @@ let _ =
         restart_return_code ;
       exit restart_return_code
 
-let pool_secret = ref (SecretString.of_string "")
+(* - this will usually be a singleton list.
+   - during pool secret rotation it will contain multiple elements,
+     but the tail will be dropped when it has completed.
+   - the head is always the pool secret that should be sent in requests *)
+let pool_secrets : SecretString.t list ref = ref []
+
+let pool_secret () = List.hd !pool_secrets
```

Comment:
Please avoid `List.hd` and instead raise a descriptive error if the list is empty.